lightning/src/ln/onion_route_tests.rs

```diff
@@ -362,7 +373,7 @@ fn test_onion_failure() {
 		let session_priv = SecretKey::from_slice(&[3; 32]).unwrap();
 		let onion_keys = onion_utils::construct_onion_keys(&Secp256k1::new(), &route.paths[0], &session_priv).unwrap();
 		msg.reason = onion_utils::build_first_hop_failure_packet(&onion_keys[0].shared_secret[..], NODE|2, &[0;0]);
-	}, ||{}, true, Some(NODE|2), Some(NetworkUpdate::NodeFailure{node_id: route.paths[0][0].pubkey, is_permanent: false}));
+	}, ||{}, true, Some(NODE|2), Some(NetworkUpdate::NodeFailure{node_id: route.paths[0][0].pubkey, is_permanent: false}), None);
```

Comment:
Same here - `temporary_node_failure` *definitely* needs an SCID to blame, otherwise we'll probably retry the same path through a node we dont know how to talk to.

Reply:
It definitely applies to BADONION as well! Good catch. It should apply any time the payment is retryable.

Reply:
Yea, we may need to debate changing that name, but the docs basically say "you can retry unless this is set"

Reply:
Added handling of `BADONION` but couldn't figure out a way to test this. 😕 

Will update the docs but I'm afraid the logic is complicated enough that I'm not certain `short_channel_id` is always set when `process_onion_failure` returns that it is retryable. Seems there are a few test cases where it is true but we expect `None`, in fact.